src/commands/cli/release/build.ts

```diff
@@ -0,0 +1,239 @@
+      char: 'd',
+      aliases: ['rctag'],
+      deprecateAliases: true,
+      exactlyOne: ['start-from-npm-dist-tag', 'start-from-github-ref'],
+    }),
+    'start-from-github-ref': Flags.string({
+      summary: messages.getMessage('flags.startFromGithubRef'),
+      char: 'g',
+      exactlyOne: ['start-from-npm-dist-tag', 'start-from-github-ref'],
+    }),
+    'build-only': Flags.boolean({
+      summary: messages.getMessage('flags.buildOnly'),
+      default: false,
+    }),
+    resolutions: Flags.boolean({
+      summary: messages.getMessage('flags.resolutions'),
+      default: true,
+      allowNo: true,
+    }),
+    only: arrayWithDeprecation({
+      summary: messages.getMessage('flags.only'),
+    }),
+    'pinned-deps': Flags.boolean({
+      summary: messages.getMessage('flags.pinnedDeps'),
+      default: true,
+      allowNo: true,
+    }),
+    patch: Flags.boolean({
+      summary: messages.getMessage('flags.patch'),
+      exclusive: ['prerelease'],
+    }),
+    prerelease: Flags.string({
+      summary: messages.getMessage('flags.prerelease'),
+      exclusive: ['patch'],
+    }),
+    snapshot: Flags.boolean({
+      summary: messages.getMessage('flags.snapshot'),
+    }),
+    schema: Flags.boolean({
+      summary: messages.getMessage('flags.schema'),
+    }),
+  };
+
+  public async run(): Promise<void> {
+    // I could not get the shelljs.exec config { fatal: true } to actually throw an error, but this works :shrug:
+    set('-e');
+
+    let auth: string;
+    const { flags } = await this.parse(build);
+
+    if (flags.prerelease === 'true' || flags.prerelease === 'false') {
+      throw new SfError(
+        'The prerelease flag is not a boolean. It should be the name of the prerelease tag, examples: dev, alpha, beta'
+      );
+    }
+
+    const pushChangesToGitHub = !flags['build-only'];
+
+    if (pushChangesToGitHub) {
+      auth = ensureString(
+        new Env().getString('GH_TOKEN') ?? new Env().getString('GITHUB_TOKEN'),
+        'The GH_TOKEN env var is required to push changes to GitHub. Use the --build-only flag to skip GitHub operations (a manual push will then be needed)'
+      );
+    }
+
+    const { ['start-from-npm-dist-tag']: startFromNpmDistTag, ['start-from-github-ref']: startFromGithubRef } = flags;
+
+    let ref: string;
+
+    if (startFromGithubRef) {
+      this.log(`Flag '--start-from-github-ref' passed, switching to '${startFromGithubRef}'`);
+
+      ref = startFromGithubRef;
+    } else {
+      this.log(`Flag '--start-from-npm-dist-tag' passed, looking up version for ${startFromNpmDistTag}`);
+
+      // Classes... I wish this was just a helper function.
+      const temp = await PackageRepo.create({ ux: new Ux({ jsonEnabled: this.jsonEnabled() }) });
+      const version = temp.package.getDistTags(temp.package.packageJson.name)[startFromNpmDistTag];
+
+      ref = version;
+    }
```

Comment:
I think this whole section should be a separate method that returns the ref

const ref = await this.getRef()